common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java

```diff
@@ -38,6 +37,23 @@ static boolean isNotEmpty(String value) {
         return Optional.ofNullable(System.getProperty(name));
     }
 
+    static Optional<String> firstSysPropertyKeyStartingWith(String keyPrefix, ProviderFactory providers) {
+        Optional<String> key = firstKeyStartingWith(keyPrefix, System.getProperties());
```

Comment:
Just to confirm: checking `System.getProperties()` is okay for config cache?
I think a javadoc on this method might be helpful to customers modifying this plugin for their own needs.

Reply:
> Just to confirm: checking System.getProperties() is okay for config cache?

Surprisingly, yes. I don't know what kind of magic the Configuration Cache is using to detect when there is undeclared access to system properties, but it does appear to be able to determine that there isn't any undeclared access going on.  Maybe it intercepts calls to System.getProperties().get() or something.

Reply:
> I think a javadoc on this method might be helpful to customers modifying this plugin for their own needs.

Good idea!